pkg/endpoint/restore.go

```diff
@@ -98,40 +77,18 @@ func ReadEPsFromDirNames(ctx context.Context, owner regeneration.Owner, policyGe
 		epDir := filepath.Join(basePath, epDirName)
 		isHost := hasHostObjectFile(epDir)
 
-		cHeaderFile := filepath.Join(epDir, common.CHeaderFileName)
 		scopedLog := log.WithFields(logrus.Fields{
 			logfields.EndpointID: epDirName,
-			logfields.Path:       cHeaderFile,
+			logfields.Path:       epDir,
 		})
-		// This function checks the presence of a bug previously observed: the
-		// file was sometimes only found after the second check.
-		// We can remove this if we haven't seen the issue occur after a while.
-		headerFileExists := func() error {
```

Comment:
Did we test for this comment ? Is it safe to remove this dual check for file.

Reply:
Take a look at the commit message. I think this check was and has been bogus: `stat() followed by open()` is racy and doesn't provide any advantages over simply doing `open()`. The open  call will return `ErrNotExist` or similar just like stat does. This is what the new code does.